src/compiler/diagnosticMessages.json

```diff
@@ -1652,6 +1652,10 @@
         "category": "Error",
         "code": 2525
     },
+    "'this' type is available only in a non-static member of a class or interface.": {

```

Comment:
Then perhaps just make it `A 'this' type is only allowed...` because that follows the intended naming conventions.